File: benchmarks/array_sum/array_sum.py
import sys
import time

n = int(sys.argv[1]) if len(sys.argv) > 1 else 10000000

start = time.perf_counter()
arr = list(range(n))
total = sum(arr)
elapsed = (time.perf_counter() - start) * 1000
print(f"TIME_MS:{elapsed:.2f}", file=sys.stderr)

print(total)

File: benchmarks/array_sum/array_sum.rb
n = ARGV[0] ? ARGV[0].to_i : 10000000

start = Process.clock_gettime(Process::CLOCK_MONOTONIC)

arr = (0...n).to_a
total = arr.sum

elapsed = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - start) * 1000
STDERR.puts "TIME_MS:#{format('%.2f', elapsed)}"
puts total
